### PR TITLE
feat: support PyTorch Profiler in DeepSpeed trials [MLG-1095]

### PR DESCRIPTION
**Pull request by azhou-determined**
## Description

Support profiling for DeepSpeed trials. DeepSpeed supports PyTorch Profiler, which we already provide an API for in PyTorch trials. Implementation for DeepSpeed is the same.

This PR is in a draft status because I am still testing this.

## Test Plan

Submit a Deepspeed trial. In the trial's init, configure the profiler:

```
   class MyTrial(DeepSpeedTrial):
       def __init__(self, context: DeepSpeedTrialContext) -> None:
           self.context = context
           ...
           self.context.set_profiler(
                activities=[
                    torch.profiler.ProfilerActivity.CPU,
                    torch.profiler.ProfilerActivity.CUDA,
                ],
                schedule=torch.profiler.schedule(
                    wait=1,
                    warmup=1,
                    active=2
                ),
            )
```

In the Web UI, open Tensorboard. There should be a "Pytorch Profiler" tab with profiling metrics.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/model-dev-guide/api-guides/apis-howto/deepspeed/deepspeed.rst [hunk 1]
```diff
@@ -303,6 +303,52 @@ interleaving micro batches:
        loss = self.model_engine.eval_batch()
        return {"loss": loss}
 
+********************
+ Step 4 - Profiling
```
Comment (ioga) [inline docs/model-dev-guide/api-guides/apis-howto/deepspeed/deepspeed.rst:307]: I am not sure it's really a "step" in the same way as the previous few?
```suggestion
 Profiling
```

File: docs/model-dev-guide/api-guides/apis-howto/deepspeed/deepspeed.rst [hunk 2]
```diff
@@ -303,6 +303,52 @@ interleaving micro batches:
...
+********************
+
+Deepspeed experiments can be profiled using PyTorch Profiler and results will automatically be
+uploaded to Tensorboard (accessible via the Determined UI). To configure profiling, call
```
Comment (ioga) [inline docs/model-dev-guide/api-guides/apis-howto/deepspeed/deepspeed.rst:311]:
```suggestion
uploaded to TensorBoard (accessible via the Determined UI). To configure profiling, call
```

**Review by ioga — APPROVED** [review]
lgtm
